clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -10172,6 +10172,23 @@ def err_bounds_type_annotation_lost_checking : Error<
     "%select{assignment|decrement|increment|initialization|statement}0">,
     InGroup<CheckBoundsDeclsChecked>;
 
+  def error_bounds_declaration_unprovable : Error<
+    "it is not possible to prove that the inferred bounds of %1 "
+    "imply the declared bounds of %1 after "
+    "%select{assignment|decrement|increment|initialization|statement}0">;
+
+  def note_free_variable_decl_or_inferred : Note<
+    "the %select{declared|inferred}0 %select{lower |upper |}1bounds use the value "
+    "of the variable '%2', and there is no relational information between '%2' "
+    "and any of the variables used by the %select{inferred|declared}0 "
```

Comment:
This may need to be reworded.  There is no relational information involving v and any of the expressions used by the other bounds.  Perhaps something along the lines of 
```
"the %select{declared|inferred}0 %slect{lower | upper |}bound use the variable %2' and there is no relational information involving '%2' and any of expressions used by the %select{inferred|declared}0 "